boskos/boskos.go

```diff
@@ -0,0 +1,295 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+)
+
+var configPath = flag.String("config", "projects.json", "Path to init projects file")
+var storage = flag.String("storage", "", "Path to presistent volume to load the state")
+
+type GCPProject struct {
+	Name     string        `json:"name"`
+	Duration time.Duration `json:"duration"`
+	Start    time.Time     `json:"start"`
+}
+
+func main() {
+	flag.Parse()
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+
+	r, err := NewRancho(*configPath, *storage)
+	if err != nil {
+		logrus.WithError(err).Fatal("Fail to create my ranch!")
+	}
+
+	http.Handle("/", handleDefault())
+	http.Handle("/request", handleRequestProject(r))
+	http.Handle("/return", handleReturnProject(r))
+	http.Handle("/sync", handleHeartbeat(r))
+
+	logrus.Info("Start Service")
+	logrus.WithError(http.ListenAndServe(":8080", nil)).Fatal("ListenAndServe returned.")
+}
+
+func handleDefault() http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		logrus.WithField("handler", "handleDefault").Infof("From %v", req.RemoteAddr)
+	}
+}
+
+func handleRequestProject(r *RanchoSantaFe) http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		logrus.WithField("handler", "handleRequestProject").Infof("From %v", req.RemoteAddr)
+
+		// validate vars
+		job := req.URL.Query().Get("job")
+		durationStr := req.URL.Query().Get("duration")
+		if job == "" || durationStr == "" {
+			logrus.Warning("[BadRequest]Job and duration must present in the request")
+			http.Error(res, "Job and duration must present in the request", http.StatusBadRequest)
+			return
+		}
+
+		duration, err := time.ParseDuration(durationStr)
+		if err != nil {
+			http.Error(res, "Invalid Duration", http.StatusBadRequest)
+			logrus.WithError(err).Errorf("Bad duration: %v", durationStr)
+			return
+		}
+
+		logrus.Infof("Job %v request for a project for %v", job, durationStr)
+
+		proj := r.GimmeCattle(job, duration)
```

Comment:
I would prefer we use a more professional verb than gimme (which many would argue is not a real word)